frontend/src/hooks/useTypingEffect.test.ts

```diff
@@ -0,0 +1,150 @@
+import { renderHook, act } from "@testing-library/react";
+import { useTypingEffect } from "./useTypingEffect";
+
+jest.useFakeTimers();
+
+describe("useTypingEffect", () => {
+  // This test fails because the hook improperly handles this case.
+  it.skip("should handle empty strings array", () => {
```

Comment:
Please note that the frontend crashes when an empty array is passed as an argument (as observed when running the first test of this file)